ygot/struct_validation_map.go

```diff
@@ -527,8 +546,26 @@ func DeepCopy(s GoStruct) (GoStruct, error) {
 	return n.Interface().(GoStruct), nil
 }
 
+// hasMergeStructOpt extracts a MergeStructOpt from the opts slice provided. In
```

Comment:
Since this is only used in a single place - I'd suggest having a single helper function that specifies `fieldOverwriteEnabled` and returns the `bool` value itself (returning false if this function would have returned `nil`).